apps/server/src/modules/tldraw/service/tldraw.ws.service.ts

```diff
@@ -2,52 +2,81 @@ import { Injectable } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import WebSocket from 'ws';
 import { applyAwarenessUpdate, encodeAwarenessUpdate, removeAwarenessStates } from 'y-protocols/awareness';
-import { encoding, decoding, map } from 'lib0';
+import { decoding, encoding, map } from 'lib0';
 import { readSyncMessage, writeSyncStep1, writeUpdate } from 'y-protocols/sync';
-import { Persitence, WSConnectionState, WSMessageType } from '../types';
+import { Buffer } from 'node:buffer';
+import { Redis } from 'ioredis';
+import { Logger } from '@src/core/logger';
+import { applyUpdate } from 'yjs';
+import {
+	RedisGeneralErrorLoggable,
+	RedisPublishErrorLoggable,
+	WebsocketCloseErrorLoggable,
+	WebsocketMessageErrorLoggable,
+	WsSharedDocErrorLoggable,
+} from '../loggable';
 import { TldrawConfig } from '../config';
-import { WsSharedDocDo } from '../domain/ws-shared-doc.do';
+import { AwarenessConnectionsUpdate, WSConnectionState, WSMessageType } from '../types';
+import { WsSharedDocDo } from '../domain';
 import { TldrawBoardRepo } from '../repo';
 import { MetricsService } from '../metrics';
 
 @Injectable()
 export class TldrawWsService {
 	public pingTimeout: number;
 
-	public persistence: Persitence | null = null;
+	private gcEnabled: boolean;
 
 	public docs = new Map();
 
+	public readonly sub: Redis;
+
+	private readonly pub: Redis;
+
 	constructor(
 		private readonly configService: ConfigService<TldrawConfig, true>,
 		private readonly tldrawBoardRepo: TldrawBoardRepo,
+		private readonly logger: Logger,
 		private readonly metricsService: MetricsService
 	) {
+		this.logger.setContext(TldrawWsService.name);
 		this.pingTimeout = this.configService.get<number>('TLDRAW_PING_TIMEOUT');
-	}
+		this.gcEnabled = this.configService.get<boolean>('TLDRAW_GC_ENABLED');
+		const redisUri: string = this.configService.get<string>('REDIS_URI');
+
+		if (!redisUri) {
+			throw new Error('REDIS_URI is not set');
+		}
+
+		this.sub = new Redis(redisUri, {
```

Comment:
Adding something with new in constructor bring you into the situation that the code is really hard to test. 
In low case you can pass the new Redis as parameter input, better by services a injectable. 
In your case you want specfic redis with fix options. For this you can create a injectable factory. Inject this factory and use it in constructor for the creation. 
Than you can inject for test a mock or what you want.


Edit: But i really expect that a TtdrawRedisService should be exists that hold all methods and stuff that is needed for redis and should be inject in this service. This also increase the readability of the code.


We can make pear programming if you like for the splitting.